### PR TITLE
apcupsd: add powerfail logic at shutdown

### PR DESCRIPTION
**Pull request by basploeger**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by basploeger** [comment]
I asked a [question on reddit](https://www.reddit.com/r/voidlinux/comments/1gwf8hy/col_command_not_found_in_musl_ci_build/) about the CI build failure for `musl`.

File: srcpkgs/apcupsd/files/apcupsd-powerfail.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+if [ -f "${PWRFAILDIR}/powerfail" ]; then
+   echo
+   echo "APCUPSD will now power off the UPS"
+   echo
+   /etc/apcupsd/apccontrol killpower
+   echo
+   echo "Please ensure that the UPS has powered off before rebooting"
+   echo "Otherwise, the UPS may cut the power during the reboot!!!"
+   echo
```
Comment (classabbyamp) [inline srcpkgs/apcupsd/files/apcupsd-powerfail.sh:12]: use `msg` (from `/etc/runit/functions`) and we shouldn't have all the extra blank lines

```
msg "Powering off UPS with APCUPSD..."
```
not sure the other message is necessary

File: srcpkgs/apcupsd/files/apcupsd/run [hunk 2]
```diff
@@ -1,4 +1,9 @@
 #!/bin/sh
 exec 2>&1
-[ -r conf] && . ./conf
+
+PWRFAILDIR=$( awk '$1 == "PWRFAILDIR" {print $2; exit}' /etc/apcupsd/apcupsd.conf )
+PWRFAILDIR=${PWRFAILDIR:=/etc/apcupsd}
```
[the remark this answers is not in the document]
Comment (classabbyamp) [reply srcpkgs/apcupsd/files/apcupsd/run:5]: also quote variable/subshell expansions

[the remark this answers is not in the document]
Comment (basploeger) [reply srcpkgs/apcupsd/files/apcupsd/run]: Done.

File: srcpkgs/apcupsd/files/apcupsd-powerfail.sh (hunk 1 above)
Comment (basploeger) [reply srcpkgs/apcupsd/files/apcupsd-powerfail.sh:12]: Done. I agree, the user is unlikely able to read and act upon the other message.

File: srcpkgs/apcupsd/files/apcupsd-powerfail.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+PWRFAILDIR=$( awk '$1 == "PWRFAILDIR" {print $2; exit}' /etc/apcupsd/apcupsd.conf )
+PWRFAILDIR=${PWRFAILDIR:-/etc/apcupsd}
```
Comment (classabbyamp) [inline srcpkgs/apcupsd/files/apcupsd-powerfail.sh:2]:
```suggestion
PWRFAILDIR="$( awk '$1 == "PWRFAILDIR" {print $2; exit}' /etc/apcupsd/apcupsd.conf )"
PWRFAILDIR="${PWRFAILDIR:-/etc/apcupsd}"
```

Comment (basploeger) [reply srcpkgs/apcupsd/files/apcupsd-powerfail.sh]: Done

**Comment by basploeger** [comment]
@classabbyamp Thanks for reviewing. PTAL!

**Pull request merged**